15_totalIntegers/solution/totalIntegers-solution.spec.js

```diff
@@ -0,0 +1,31 @@
+const totalIntegers = require('./totalIntegers-solution');
+
+describe('totalIntegers', () => {
```

Comment:
I think it would be valuable to start with just a few very basic tests e.g.

```javascript
test('Works with a simple array of numbers', () => {
  expect(totalIntegers([1, 2, 3])).toBe(3);
});
test('Ignores non-integer values', () => {
  expect(totalIntegers([1, 2, '3', 4])).toBe(3);
});
test('Works with simple objects', () => {
  expect(totalIntegers({ a: 1, b: '2', c: 3 })).toBe(2);
});
```

Then afterwards we can get into the "more complex" test cases.
This helps people get the basic logic down, before then trying to work with any of the recursive sides of things.